adcc/State2States.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+## vi: tabstop=4 shiftwidth=4 softtabstop=4 expandtab
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2020 by the adcc authors
+##
+## This file is part of adcc.
+##
+## adcc is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published
+## by the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## adcc is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with adcc. If not, see <http://www.gnu.org/licenses/>.
+##
+## ---------------------------------------------------------------------
+import numpy as np
+
+from . import adc_pp
+from .misc import cached_property
+from .timings import timed_member_call
+
+from .Excitation import mark_excitation_property
+from .ElectronicTransition import ElectronicTransition
+
+
+class State2States(ElectronicTransition):
+    def __init__(self, data, method=None, property_method=None, initial=0):
+        """Construct a State2States class from some data obtained
+            from an interative solver or other suitable data.
+
+            The class provides access to the results from an ADC calculation
+            as well as derived properties. Properties are computed lazily
+            on the fly as requested by the user.
+
+            By default the ADC method is extracted from the data object
+            and the property method in property_method is set equal to
+            this method, except ADC(3) where property_method=="adc2".
+            This can be overwritten using the parameters.
```

Comment:
I don't think I would just copy the docs from the ExcitedStates class here.

Reply:
Done.